rest-api/test/unit_test/offline_test/public_metrics_export_test.py

```diff
@@ -0,0 +1,261 @@
+    if gender_code:
+      code_answers.append(('genderIdentity', Concept(PPI_SYSTEM, gender_code)))
+    if date_of_birth:
+      date_answers.append(('dateOfBirth', date_of_birth))
+    if state:
+      code_answers.append(('state', Concept(PPI_SYSTEM, state)))
+    qr = make_questionnaire_response_json(
+        participant_id,
+        questionnaire_id,
+        code_answers=code_answers,
+        date_answers=date_answers)
+    self.send_post('Participant/%s/QuestionnaireResponse' % participant_id, qr)
+
+  def _submit_consent_questionnaire_response(
+      self, participant_id, questionnaire_id, ehr_consent_answer):
+    code_answers = [('ehrConsent', Concept(PPI_SYSTEM, ehr_consent_answer))]
+    qr = make_questionnaire_response_json(
+        participant_id, questionnaire_id, code_answers=code_answers)
+    self.send_post('Participant/%s/QuestionnaireResponse' % participant_id, qr)
+
+  def _submit_empty_questionnaire_response(self, participant_id,
+                                           questionnaire_id):
+    qr = make_questionnaire_response_json(participant_id, questionnaire_id)
+    self.send_post('Participant/%s/QuestionnaireResponse' % participant_id, qr)
+
+  def _create_data(self):
+    HPODao().insert(HPO(hpoId=PITT_HPO_ID + 1, name='AZ_TUCSON'))
+    HPODao().insert(HPO(hpoId=PITT_HPO_ID + 2, name='TEST'))
+    SqlTestBase.setup_codes(
+        ANSWER_FIELD_TO_QUESTION_CODE.values() + [EHR_CONSENT_QUESTION_CODE],
+        code_type=CodeType.QUESTION)
+    SqlTestBase.setup_codes(
+        FIELD_TO_QUESTIONNAIRE_MODULE_CODE.values(), code_type=CodeType.MODULE)
+    # Import codes for white and female, but not male or black.
+    SqlTestBase.setup_codes(
+        [
+            RACE_WHITE_CODE, CONSENT_PERMISSION_YES_CODE,
+            RACE_NONE_OF_THESE_CODE, PMI_PREFER_NOT_TO_ANSWER_CODE,
+            CONSENT_PERMISSION_NO_CODE, 'female', 'PIIState_VA'
+        ],
+        code_type=CodeType.ANSWER)
+    participant_dao = ParticipantDao()
+
+    questionnaire_id = self.create_questionnaire('questionnaire3.json')
+    questionnaire_id_2 = self.create_questionnaire('questionnaire4.json')
+    questionnaire_id_3 = self.create_questionnaire(
+        'all_consents_questionnaire.json')
+    with FakeClock(TIME):
+      participant = Participant(
+          participantId=1,
+          biobankId=2,
+          providerLink=make_primary_provider_link_for_name('AZ_TUCSON'))
+      participant_dao.insert(participant)
+      self.send_consent('P1', email='bob@gmail.com')
+
+    with FakeClock(TIME):
+      # Participant 2 starts out unpaired; later gets paired automatically when their physical
+      # measurements come in.
+      participant2 = Participant(participantId=2, biobankId=3)
+      participant_dao.insert(participant2)
+      self.send_consent('P2', email='bob@fexample.com')
+
+    with FakeClock(TIME):
+      # Test HPO affiliation; this test participant is ignored.
```

Comment:
It looks like you copy / pasted this from the metrics_export_test code?

We could instead refactor it to a common place... or modify it. Some of the stuff (e.g. what time things happen) doesn't matter much for your metrics.

Note these comments indicate what *should* happen WRT test participants in metrics... but I'm not sure they are.